### DISCUSSION TITLE
Add parameters to dataset files

### DISCUSSION THREAD

**Discussion by RConstruktiv** [coqui-ai/TTS#2644] [Ideas]
Just a general idea to deal with emotions and general dynamic of a voice. I couldn't find anyone suggesting it.

What if we could add emotion parameters to the dataset transcript, so the trained model knows exactly what for example a sad voice sounds like. Then you can add the paramters before a sentence with brackets like right now. This way you could maybe add sounds like laughing.

Just a quick thought.

**Comment by Alienpups** [comment]
Really like that idea... would add a lot of voice dynamic...

**Comment by erogol** [comment]
I like the idea. Do you have an idea how we can create such a dataset?

**Reply by RConstruktiv** [reply]
We could either add it to the transcript with [happy] or create a folder structure so you create a dataset for each mood.

**Comment by thorstenMueller** [comment]
You mean, like SSML. So tags inside the transcript - like: <neutral>The sun is shining</neutral><happy>This makes me happy</happy>?
If we find a syntax for that maybe i can record a new open voice dataset with mixed emotions and tagged this (or a similar) way.

**Reply by erogol** [reply]
I am not sure if SSML is the right approach but love creating a dataset 👍

**Reply by thorstenMueller** [reply]
Okay, i'll keep an eye on this discussion and once there's a structure defined i'll check on recording a new "german" dataset 😊.

**Comment by manmay-nakhashi** [comment]
Probably we can add it like [laughter] and give it a special token to it we can make a dictionary for non vocal tokens(ids), when ever [] comes we don't tokenize it how we should be.

**Reply by erogol** [reply]
Makes sense however, it looks hard to annotate a dataset like this. IS there any ASR model that can do the job you know?

**Reply by manmay-nakhashi** [reply]
What if we do it with vc model ? 🤔 There is a non vocal dataset which contains this sounds , apply vc on it and create a larger dataset and while training we augment it, or make a classifier out of the data, for labeling.
